rfd/9999-passwordless.md

```diff
@@ -0,0 +1,501 @@
+            │                      │                                             │
+            │                      │                1.1. challenge               │
+            │                      │ <────────────────────────────────────────────
+            │                      │                                             │
+            │ 2. list credentials  │                                             │
+            │<──────────────────────                                             │
+            │                      │                                             │
+            │                      │                                             │
+            │──────────────────────>                                             │
+            │                      │                                             │
+            │2.1. choose credential│                                             │
+            │<──────────────────────                                             │
+            │                      │                                             │
+            │                      │                                             │
+            │──────────────────────>                                             │
+            │                      │                                             │
+            │ 2.2. sign challenge  │                                             │
+            │<──────────────────────                                             │
+            │                      │                                             │
+            │2.3. signed challenge │                                             │
+            │──────────────────────>                                             │
+            │                      │                                             │
+            │                      │ 3. Authenticate(userHandle, signedChallenge)│
+            │                      │ ────────────────────────────────────────────>
+            │                      │                                             │
+            │                      │   3.1. signed certificates or web session   │
+            │                      │ <────────────────────────────────────────────
+     ┌──────┴──────┐              user                                       ┌───┴────┐
+     │authenticator│               ┌─┐                                       │Teleport│
+     └─────────────┘               ║"│                                       └────────┘
+                                   └┬┘
+                                   ┌┼┐
+                                    │
+                                   ┌┴┐
+```
+
+Authentication looks similar to our well-known challenge/response protocols,
+with the exception that the initial challenge is created _without_ knowledge of
+the user identity. This is a departure from the current authn model where
+[challenge generation is gated on username and password](
```

Comment:
> challenge is created _without_ knowledge of the user identity.

Do we need to worry about an abuse of requests to this backend public endpoint that creates challenges for anonymous users?

Reply:
I talk a bit about this in the Security section. 

I think, in the end, this is an inherent problem for authentication endpoints - this one is not that different. We can (and should!) rate limit access to mitigate bad behavior, but there isn't a whole lot to do other than that. (I'm open to ideas if you have any.)

Reply:
I think the difference with existing public auth endpoints is that this one introduces a side affect as it creates a challenge (stored in the memory/db). So one idea would be to use a circular in-memory buffer with a reasonable size (in addition to rate limits) for storing anonymous login state.

Reply:
Noted - thanks for the suggestion. I'll add a nod to in the design later.

Reply:
Done.